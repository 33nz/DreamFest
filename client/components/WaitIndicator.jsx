import React from 'react'
import { connect } from 'react-redux'

function WaitIndicator ({ waiting }) {
  return waiting
    ? <p>WAITING</p>
    : <br/>
}

function mapStateToProps (state) {
  return {
    waiting: state.waiting
  }
}

export default connect(mapStateToProps)(WaitIndicator)
