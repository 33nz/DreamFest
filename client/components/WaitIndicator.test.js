import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { WaitIndicator } from './WaitIndicator'

test('renders wait indicator when waiting is true', () => {
  const { asFragment } = render(<WaitIndicator waiting={true} />)
  expect(asFragment()).toMatchSnapshot()
})

test('renders nothing when waiting is false', () => {
  const { asFragment } = render(<WaitIndicator waiting={false} />)
  expect(asFragment()).toMatchSnapshot()
})
