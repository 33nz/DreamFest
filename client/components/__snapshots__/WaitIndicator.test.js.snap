// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing when waiting is false 1`] = `<DocumentFragment />`;

exports[`renders wait indicator when waiting is true 1`] = `
<DocumentFragment>
  <img
    class="wait-indicator"
    src="/animated-circle.gif"
  />
</DocumentFragment>
`;
